src/Control/Effect/Carrier.hs

```diff
@@ -34,46 +31,109 @@ class HFunctor h where
 --   1. Be functorial in their last two arguments, and
 --   2. Support threading effects in higher-order positions through using the carrier’s suspended state.
 --
--- All first-order effects (those without recursive occurrences of @m@) admit a default definition
--- of 'handle'. The @-XDeriveAnyClass@ extension allows derivation of both 'HFunctor' and 'Effect':
```

Comment:
I removed this part of the note because `DeriveAnyClass` is only necessary to name the classes in `deriving` clauses, not to actually derive the instance. E.g. this with `-XDeriveAnyClass`:

```haskell
data State s m k
  = Get (s -> m k)
  | Put s (m k)
  deriving (Functor, Generic1, HFunctor, Effect)
```

is equivalent to this without:

```haskell
data State s m k
  = Get (s -> m k)
  | Put s (m k)
  deriving (Functor, Generic1)

instance HFunctor (State s)
instance Effect (State s)
```

Users don’t have to do anything special to use the latter, so I didn’t feel it was worth calling out in such a prominent place.